core/controllers/classifier.py

```diff
@@ -79,7 +79,10 @@ def verify_signature(message, vm_id, received_signature):
     if secret is None:
         return False
 
-    generated_signature = generate_signature(secret, message)
+    if message is None:
```

Comment:
Not sure who you mean by "your", but yes -- I stand by my previous comment in this thread. The signature stuff should be transparent and part of the infrastructure; it should not need to know the context of any requests/responses.